sound/soc/sof/sof-of-dev.c

```diff
@@ -113,6 +117,8 @@ static int sof_of_probe(struct platform_device *pdev)
 static int sof_of_remove(struct platform_device *pdev)
 {
 	pm_runtime_disable(&pdev->dev);
+	pm_runtime_set_suspended(&pdev->dev);
+	pm_runtime_put_noidle(&pdev->dev);
```

Comment:
are these last two needed?
Specifically the put_noidle was only needed for PCI, we don't use it anywhere else, do we?

Reply:
I will have a deeper look at this tommorrow. I've seen this pattern in multiple drivers. To be honest, I dont understand PM really well.

Reply:
yup agree the last 2 shouldnt be needed and might even cause an imbalance. Just disabling PM in remove() should be good enough